Sources/SwiftSyntax/Trivia.swift

```diff
@@ -42,6 +42,68 @@ public struct Trivia: Sendable {
     pieces.isEmpty
   }
 
+  /// The string contents of all the comment pieces with any comments tokens trimmed.
+  ///
+  /// Each element in the array is the trimmed contents of a line comment, or, in the case of a multi-line comment a trimmed, concatenated single string.
+  public var commentValues: [String] {
+    var comments = [String]()
+    var partialComments = [String]()
+
+    var foundStartOfCodeBlock = false
+    var foundEndOfCodeBlock = false
+    var isInCodeBlock: Bool { foundStartOfCodeBlock && !foundEndOfCodeBlock }
+
+    for piece in pieces {
+      switch piece {
+      case .blockComment(let text), .docBlockComment(let text):
+        let text = text.trimmingCharacters(in: "\n")
+
+        foundStartOfCodeBlock = text.hasPrefix("/*")
+        foundEndOfCodeBlock = text.hasSuffix("*/")
+
+        let sanitized =
+          text
+          .split(separator: "\n")
+          .map { $0.trimmingAnyCharacters(in: "/*").trimmingAnyCharacters(in: " ") }
+          .filter { !$0.isEmpty }
```

Comment:
Does this drop empty lines in the comment? Eg. in the following? Because I don’t think we want to. For example if you interpret the comment as markdown, empty lines have meaning.

```swift
/*
Paragraph 1

Paragraph 2
*/
```